remoting/remoting-bolt/src/main/java/com/alipay/sofa/rpc/codec/bolt/SofaRpcSerialization.java

```diff
@@ -280,6 +279,47 @@ protected void recordSerializeRequest(RequestCommand requestCommand, InvokeConte
         return false;
     }
 
+    @VisibleForTesting
+    protected void parseRequestHeader(Map<String, String> headerMap, SofaRequest sofaRequest) {
+        // 处理 tracer
+        parseRequestHeader(RemotingConstants.RPC_TRACE_NAME, headerMap, sofaRequest);
+        replaceWithHeaderMap(headerMap, sofaRequest.getRequestProps());
+    }
+
+    private void parseRequestHeader(String key, Map<String, String> headerMap,
+                                    SofaRequest sofaRequest) {
+        Map<String, String> traceMap = new HashMap<String, String>();
+        CodecUtils.treeCopyTo(key + ".", headerMap, traceMap, true);
+        Object traceCtx = sofaRequest.getRequestProp(key);
+        if (traceCtx == null) {
+            sofaRequest.addRequestProp(key, traceMap);
+        } else if (traceCtx instanceof Map) {
+            ((Map<String, String>) traceCtx).putAll(traceMap);
+        }
+    }
+
+    private void replaceWithHeaderMap(Map<String, String> headerMap, Map requestProps) {
+        if (headerMap == null) {
+            return;
+        }
+        if (requestProps == null) {
+            requestProps = new HashMap<String, Object>();
+        }
+        // 1. 如果 key 已经在requestProps存在，value 为 String 时进行覆盖
+        // 2. header 中的 value 为 Blank 时不进行覆盖
+        for (Map.Entry<String, String> entry : headerMap.entrySet()) {
+            Object o = requestProps.get(entry.getKey());
+            if (o == null) {
+                requestProps.put(entry.getKey(), entry.getValue());
+            } else if (o instanceof String) {
+                if (StringUtils.isBlank((CharSequence) o)
```

Comment:
What is the purpose of converting a `String` to a `CharSequence` here?

Reply:
The parameter type of the isBlank method is CharSequence.